app/views/declaration/summary/package_information.scala.html

```diff
@@ -15,42 +15,79 @@
  *@
 
 @import forms.declaration.PackageInformation
-@import models.requests.JourneyRequest
-@import models.viewmodels.HtmlTableRow
 @import models.Mode
+@import models.requests.JourneyRequest
+@import uk.gov.hmrc.govukfrontend.views.html.components._
+@import views.html.components.gds.summary_list
+@import views.declaration.summary.TableCell
 
-@(mode: Mode, itemId: String, itemNo: Int, packageInformation: Seq[PackageInformation])(implicit messages: Messages, journeyRequest: JourneyRequest[_])
+@this(
+        govukTable: GovukTable,
+        summaryList: summary_list
+)
 
-<table>
-@if(packageInformation.nonEmpty) {
+@(mode: Mode, itemId: String, itemNo: Int, packageInformation: Seq[PackageInformation])(implicit messages: Messages, journeyRequest: JourneyRequest[_])
 
-    <caption class="heading-small"><span id="package-information-@itemNo">@messages("declaration.summary.items.item.packageInformation")</span></caption>
-    <tr class="section">
-        <td class="bold"><span id="package-information-type-@itemNo">@messages("declaration.summary.items.item.packageInformation.type")</span></td>
-        <td class="bold"><span id="package-information-number-@itemNo">@messages("declaration.summary.items.item.packageInformation.number")</span></td>
-        <td class="bold"><span id="package-information-markings-@itemNo">@messages("declaration.summary.items.item.packageInformation.markings")</span></td>
-        <td></td>
-    </tr>
-    @packageInformation.zipWithIndex.map { case (information, index) =>
-    <tr class="section">
-        <td class="exports-table__cell"><span id="package-information-@itemNo-code-@index">@information.typesOfPackagesText.getOrElse("")</span></td>
-        <td class="exports-table__cell"><span id="package-information-@itemNo-number-@index">@information.numberOfPackages.getOrElse("")</span></td>
-        <td class="exports-table__cell"><span id="package-information-@itemNo-marks-@index">@information.shippingMarks.getOrElse("")</span></td>
-        <td class="exports-table__cell exports-table__cell--numeric">
-            <a href="@controllers.declaration.routes.PackageInformationController.displayPage(mode, itemId)" id="package-information-@itemNo-change-@index">
-                <span aria-hidden="true">@messages("site.change")</span>
-                <span class="visually-hidden">@messages("declaration.summary.items.item.packageInformation.change", information.typesOfPackagesText, information.shippingMarks, itemNo)</span>
+@changeLink(information: PackageInformation) = {
+    <a class="govuk-link" href="@controllers.declaration.routes.PackageInformationController.displayPage(mode, itemId)">
```

Comment:
Can we extract link with visually hidden parts as component or default non-showing parameter in link ?

Reply:
You mean extract the "changeLink" method into a component?   That method is copy/pasted a few times in the PR so that would be worth-while